### PR TITLE
Update decorations for global variables

### PR DESCRIPTION
**Pull request by vmaksimo**
This change is basically an update of #1389 for spec changes.

Implementation of the feature was based on Intel extension which was not officially published to Khronos.
Now it has been split, updated, and published to Khronos by https://github.com/KhronosGroup/SPIRV-Registry/pull/205

Summary of the things that have changed:

* Capability names and a new capability was added
* Values for decorations have been updated to enums
* Decoration names and IDs have been changed

Specs:
https://github.com/KhronosGroup/SPIRV-Registry/blob/main/extensions/INTEL/SPV_INTEL_global_variable_fpga_decorations.asciidoc https://github.com/KhronosGroup/SPIRV-Registry/blob/main/extensions/INTEL/SPV_INTEL_global_variable_host_access.asciidoc

### CODE REVIEW

**Comment by vmaksimo** [comment]
TODO:

- [ ] Actually use enum values instead of numbers for `HostAccessINTEL` and `InitModeINTEL`

**Comment by vmaksimo** [comment]
Please take a look @MrSidims @asudarsa @LU-JOHN @bwlodarcz

File: test/transcoding/global_var_decorations.ll [hunk 1]
```diff
@@ -0,0 +1,68 @@
+; RUN: llvm-as %s -o %t.bc
```
Comment (MrSidims) [inline test/transcoding/global_var_decorations.ll:1]: May I ask you to move this test to a new directory if appropriate?

**Review by MrSidims — COMMENTED** [review]
LGTM, just a minor suggesting

File: test/transcoding/global_var_decorations.ll (hunk 1 above)
Comment (vmaksimo) [reply test/transcoding/global_var_decorations.ll]: Applied

**Review by MrSidims — APPROVED** [review]

**Pull request merged**